// Copyright (C) 2012 Cory Maccarrone <darkstar6262@gmail.com>

option cc_generic_services = true;

package backend;

message EmptyMessage {}

message BackupSet {
  optional string name = 1;
}

message EnumerateBackupSetsResponse {
  repeated BackupSet backup_sets = 1;
}

service BtrfsBackendService {
  // Initiate a ping to verify the service is alive.
  rpc Ping(EmptyMessage) returns (EmptyMessage);

  // Get all the backup sets managed by the backend.
  rpc EnumerateBackupSets(EmptyMessage) returns (EnumerateBackupSetsResponse);
}

